.travis.yml

```diff
@@ -79,6 +79,15 @@ before_install:
     - which python; python --version
     - tools/build_versions.py
     - tools/check_sdist.py
+    # It is really annoying if flake8 causes your tests to fail first
+    # but in the event that you have a typo, flake8 is pretty good at
+    # catching it.
+    # Therefore, it is best to run flake8 first, on a single build
+    # with the oldest version of python we support
+    - if [[ $TEST_FLAKE8 == "1" ]]; then
```

Comment:
Should all of this be in a dedicated `tools\flake8.py`?

Reply:
done. `flake8.sh`